components/screens/deliveries/NewDelivery.tsx

```diff
@@ -305,19 +322,26 @@ class NewDelivery extends React.Component<Props, State> {
    * Handles product change
    */
   private handleProductChange = async (productId: string) => {
-    if (!this.props.accessToken) {
+    const { accessToken } = this.props;
+
+    if (!accessToken) {
       return;
     }
+
     this.setState({ productId });
     const products = this.state.products;
     const product = products.find((product) => product.id === productId)
     const Api = new PakkasmarjaApi();
-    const productPricesService = await Api.getProductPricesService(this.props.accessToken.access_token);
-    const productPrice: ProductPrice[] = await productPricesService.listProductPrices(productId, "CREATED_AT_DESC", undefined, undefined, 1);
+    const productPricesService = Api.getProductPricesService(accessToken.access_token);
+    const deliveryQualitiesService = Api.getDeliveryQualitiesService(accessToken.access_token);
+    const [ productPrice, deliveryQualities ] = await Promise.all([
+      productPricesService.listProductPrices(productId, "CREATED_AT_DESC", undefined, undefined, 1),
+      (await deliveryQualitiesService.listDeliveryQualities(ItemGroupCategory.FRESH, productId)).reverse()
```

Comment:
sort instead by descending priceBonus amount

Reply:
Reversing array removed from this line